capability.md

```diff
@@ -0,0 +1,60 @@
+Capability
+==========
+
+Capability Query
+---
+
+The capability query provides a list of all modes supported by each pin and the resolution used by each pin. Each pin has 2 bytes for each supported mode and a value of 127 to mark the end of that pin's data. The number of pins supported is inferred by the message length.
+
+### Capabilities query
+```c++
+0  START_SYSEX              (0xF0)
+1  CAPABILITY_QUERY         (0x6B)
+2  END_SYSEX                (0xF7)
+```
+
+### Capabilities response
+```c++
+0  START_SYSEX              (0xF0)
+1  CAPABILITY_RESPONSE      (0x6C)
+2  1st supported mode of pin 0
+3  1st mode's resolution of pin 0
+4  2nd supported mode of pin 0
+5  2nd mode's resolution of pin 0
+... additional modes/resolutions, followed by PIN_MODE_IGNORE,
```

Comment:
The role of PIN_MODE_IGNORE is actually to prevent the client from modifying specific pins. If a pin has no modes to report via a capability query the only value returned for that pin will be 0x7F, which doesn't necessarily mean that pin is ignored, but that it has no capabilities defined. This is the case for pins 0 and 1 on an Arduino Uno for example since those pins are connected to the serial transport that Firmata uses.